api/lambdas/sqs-to-fhir/index.js

```diff
@@ -0,0 +1,335 @@
+    MessageAttributes: attributesToSend(message.messageAttributes),
+    DelaySeconds: delayWhenRetryingSeconds, // wait at least that long before retrying
+  };
+  try {
+    await sqs.sendMessage(sendParams).promise();
+  } catch (err) {
+    console.log(`Failed to re-enqueue message: `, message, err);
+    Sentry.captureException(err, {
+      extra: { message, sendParams, context: "reEnqueue" },
+    });
+  }
+}
+
+async function dequeue(message) {
+  const deleteParams = {
+    QueueUrl: sourceQueueURL,
+    ReceiptHandle: message.receiptHandle,
+  };
+  try {
+    await sqs.deleteMessage(deleteParams).promise();
+  } catch (err) {
+    console.log(`Failed to remove message from queue: `, message, err);
+    Sentry.captureException(err, {
+      extra: { message, deleteParams, context: "dequeue" },
+    });
+  }
+}
+
+async function reportMetrics(metrics) {
+  const { download, upsert, job } = metrics;
+  const metric = (name, values, serviceName) => ({
+    MetricName: name,
+    Value: parseFloat(values.duration),
+    Unit: "Milliseconds",
+    Timestamp: values.timestamp,
+    Dimensions: [{ Name: "Service", Value: serviceName ?? lambdaName }],
+  });
+  try {
+    await cloudWatch
+      .putMetricData({
+        MetricData: [
+          metric("Download", download),
+          metric("Upsert", upsert),
+          metric("Job duration", job, "FHIR Conversion Flow"),
+        ],
+        Namespace: metricsNamespace,
+      })
+      .promise();
+  } catch (err) {
+    console.log(`Failed to report metrics, `, metrics, err);
+    Sentry.captureException(err, { extra: { metrics } });
+  }
+}
+
+async function reportMemoryUsage() {
+  var mem = process.memoryUsage();
+  console.log(
+    `[MEM] rss:  ${kbToMbString(mem.rss)}, ` +
+      `heap: ${kbToMbString(mem.heapUsed)}/${kbToMbString(mem.heapTotal)}, ` +
+      `external: ${kbToMbString(mem.external)}, ` +
+      `arrayBuffers: ${kbToMbString(mem.arrayBuffers)}, `
+  );
+  try {
+    await cloudWatch
+      .putMetricData({
+        MetricData: [
+          {
+            MetricName: "Memory total",
+            Value: kbToMb(mem.rss),
+            Unit: "Megabytes",
+            Timestamp: new Date().toISOString(),
+            Dimensions: [{ Name: "Service", Value: lambdaName }],
+          },
+        ],
+        Namespace: metricsNamespace,
+      })
+      .promise();
+  } catch (err) {
+    console.log(`Failed to report memory usage, `, mem, err);
+    Sentry.captureException(err, { extra: { mem } });
+  }
+}
+
+function kbToMbString(value) {
+  return Number(kbToMb(value)).toFixed(2) + "MB";
+}
+
+function kbToMb(value) {
+  return value / 1048576;
+}
+
+async function streamToString(stream) {
+  const chunks = [];
+  return new Promise((resolve, reject) => {
+    stream.on("data", chunk => chunks.push(Buffer.from(chunk)));
+    stream.on("error", err => reject(err));
+    stream.on("end", () => resolve(Buffer.concat(chunks).toString("utf8")));
+  });
+}
```

Comment:
obviously doesn't have to be addressed now, but having these things in a common place for lambdas in the future would be great - maybe along with the ts lambda migration 😄

Reply:
💯 , I think we should go with TS lambdas in a single file, sharing a common Lambda layer with the dependencies and utility files.

https://github.com/metriport/metriport-internal/issues/715